middleware/middleware.go

```diff
@@ -16,6 +16,8 @@ import (
 // Config holds the configuration common across all middleware.
 type Config struct {
 	AnnounceInterval time.Duration `yaml:"announce_interval"`
+	MaxNumWant       uint32        `yaml:"max_numwant"`
+	DefaultNumWant   uint32        `yaml:"default_numwant"`
```

Comment:
We could put them there - that would however make it impossible for a middleware to change them (the response middleware is just gonna use the configured values).